nni/nas/oneshot/pytorch/sampling.py

```diff
@@ -92,6 +92,37 @@ def export(self) -> dict[str, Any]:
         )
         return super().export()
 
+    def _get_base_model(self):
+        assert isinstance(self.model.model, nn.Module)
+        base_model: nn.Module = self.model.model
+        return base_model
+
+    def state_dict(self, destination: Any=None, prefix: str='', keep_vars: bool=False) -> Dict[str, Any]:
+        base_model = self._get_base_model()
+        state_dict = base_model.state_dict(destination=destination, prefix=prefix, keep_vars=keep_vars)
+        return state_dict
+
+    def load_state_dict(self, state_dict, strict: bool=True) -> None:
+        base_model = self._get_base_model()
+        base_model.load_state_dict(state_dict=state_dict, strict=strict)
+
+    def sub_state_dict(self, arch: dict[str, Any], destination: Any=None, prefix: str='', keep_vars: bool=False) -> Dict[str, Any]:
+        """Given the architecture dict, return the state_dict which can be directly loaded by the fixed subnet.
+
+        Parameters
+        ----------
+        arch : dict[str, Any]
+            subnet architecture dict.
+
+        Returns
+        -------
+        dict
+            Subnet state dict.
+        """
+        self.resample(arch)
```

Comment:
the name "resample" is a little misleading, the literal meaning is sampling a subnet from space, but it receives an "arch"?

Reply:
"arch" is a model space rather than a single subnet arch?

Reply:
> "arch" is a model space rather than a single subnet arch?

`arch` is the  single subnet arch.

for example, a model space is {"embed_dim": [32, 64, 128], mlp_ratio: [3.0, 3.5, 4.0]}. then a possible subnet arch is {"embed_dim": 64, mlp_ratio: 3.5}